Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Runtime/LocalLambdaRuntime.cs

```diff
@@ -53,20 +53,17 @@ public static ILocalLambdaRuntime Initialize(string directory, IAWSService awsSe
                 throw new DirectoryNotFoundException($"Directory containing built Lambda project does not exist {directory}");
             }
 
-            var depsFile = Directory.GetFiles(directory, "*.deps.json").FirstOrDefault();
-            if (depsFile == null)
-            {
-                throw new Exception($"Failed to find a deps.json file in the specified directory ({directory})");
-            }
+            var lambdaProjectDirectory = Utils.FindLambdaProjectDirectory(directory);
```

Comment:
There are at least 2 problems with the approach:
1) The assembly name and the folder name are not guaranteed to be 1 to 1. The assembly name can be changed checking the `AssemblyName` MSBuild property. The `AssemblyName` property could be in the project file or imported in from another msbuild file.
2) There not actually be any source or a project file. It is possible to have the test tool point only the binaries.

What I would suggest is if there are multiple deps.json files inspect the JSON files to see which one is the root. For example in the provided repository https://github.com/nathanvj/LambdaTestToolError there are 2 deps.json files,  `LambdaTestToolError.API.deps.json` and `LambdaTestToolError.Lambda.deps.json` . My rough algorithm would be:
* If only one deps.json file return using that file
* Loop through each deps.json file
  * Extract the name of the assembly from the file name of the deps.json. (Remove the .deps.json suffix)
  * Loop through all of the other deps.json files
  * For each `target` in the file inspect each assembly to see if the assembly for the deps.json file being inspect for rootness is in there. If it is not referenced in any other deps.json return true other false for whether it is the root deps.json file. 
For example in the example when we inspect the `LambdaTestToolError.API.deps.json` file for being the root and then search to see if `LambdaTestToolError.API` is referenced `LambdaTestToolError.Lambda.deps.json` we would return false for being root. Looking at the JSON snippet below the `LambdaTestToolError.Lambda` assembly referenced in the deps.json file has a dependency to `LambdaTestToolError.API`

```
  "targets": {
    ".NETCoreApp,Version=v6.0": {
      "LambdaTestToolError.Lambda/1.0.0": {
        "dependencies": {
          "Amazon.Lambda.Core": "2.1.0",
          "Amazon.Lambda.Serialization.SystemTextJson": "2.3.0",
          "LambdaTestToolError.API": "1.0.0"
        },
        "runtime": {
          "LambdaTestToolError.Lambda.dll": {}
        }
      },
      "Amazon.Lambda.Core/2.1.0": {
        "runtime": {
          "lib/net6.0/Amazon.Lambda.Core.dll": {
            "assemblyVersion": "1.0.0.0",
            "fileVersion": "1.0.0.0"
          }
        }
      },
     ...
```
When checking `LambdaTestToolError.Lambda.deps.json` we would see that ``LambdaTestToolError.Lambda` is not referenced as a dependency in any other file.

Reply:
Implemented the suggested logic and added unit tests.

Also ensured that the  https://github.com/nathanvj/LambdaTestToolError project can be debugged by the updated local version of the test tool.